hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -230,12 +220,12 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
                 .collect(Collectors.toSet());
       }
 
-      Map<String, String> summary =
-          Optional.ofNullable(metadata.currentSnapshot())
-              .map(Snapshot::summary)
-              .orElseGet(ImmutableMap::of);
-      setHmsTableParameters(
-          newMetadataLocation, tbl, metadata, removedProps, hiveEngineEnabled, summary);
+      icebergTableConverter.setHmsTableParameters(newMetadataLocation, tbl,
+          metadata, removedProps, hiveEngineEnabled);
+      if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
```

Comment:
You introduced a class to populate the table props needed for HMS, but this single one for some reason is not taken care by that class.

Reply:
Yes, the reason is  it is not needed where this PR originates from. So I can reuse the table property setting part in my PR, but this part won't be a common, this is why I have it here, but not in the other PR:
https://github.com/apache/hive/pull/5628#discussion_r1979230748

Reply:
But now you have a helper class that sets the table properties needed for HMS except this one. Hard to argue why this shouldn't go into the helper class. When someone will also use the helper class, how would they know that it doesn't set everything and they have to set this one separately?

Reply:
I understand your concerns, but the currentMetadataLocation is only available in BaseMetastoreTableOperations (base class of HiveTableOperations) and the value is also calculated in BaseMetastoreTableOperations.refreshFromMetadataLocation, so can not just move it to HMSTablePropertyHelper.

Reply:
currentMetadataLocation could be a param to the new util class and then you could have a similar code to check nullness and such and setting `PREVIOUS_METADATA_LOCATION_PROP` accordingly, right?

Reply:
Done.